core/tx_list.go

```diff
@@ -506,8 +506,12 @@ func (l *txList) Filter(costLimit *uint256.Int, gasLimit uint64) (types.Transact
 				lowest = nonce
 			}
 		}
+		l.txs.m.Lock()
 		invalids = l.txs.filter(func(tx *types.Transaction) bool { return tx.Nonce() > lowest })
+		l.txs.m.Unlock()
 	}
+	l.txs.m.RLock()
+	defer l.txs.m.RUnlock()
 	l.txs.reheap()
```

Comment:
Could we remove this Rlock and use that reheap code https://github.com/maticnetwork/bor/pull/806/files#diff-4aae684b15e6df1ff1c22f6845c43bd308a67dc43fc4fdec6f620979ac6941a9R157